src/ai/backend/manager/scheduler/dispatcher.py

```diff
@@ -731,16 +731,28 @@ async def _transition() -> Sequence[PendingSession]:
                 log.debug("prepare(): preparing {} session(s)", len(scheduled_sessions))
                 async with aiotools.TaskGroup() as tg:
                     for scheduled_session in scheduled_sessions:
-                        await self.registry.event_producer.produce_event(
-                            SessionPreparingEvent(
-                                scheduled_session.session_id,
-                                scheduled_session.session_creation_id,
-                            ),
-                        )
-                        tg.create_task(self.start_session(
-                            sched_ctx,
-                            scheduled_session,
-                        ))
+                        sgroup_name = scheduled_session.scaling_group
+                        async with self.db.begin() as conn:
+                            query = (
+                                sa.select(scaling_groups.c.scheduler_opts)
+                                .select_from(scaling_groups)
+                                .where(scaling_groups.c.name == sgroup_name)
+                            )
+                            result = await conn.execute(query)
+                            row = result.first()
+                            allowed_session_type = row['scheduler_opts']['allowed_session_type']
```

Comment:
The location of this check should be moved to PENDING -> SCHEDULED transition handler, not here in SCHEDULED -> PREPARING transition handler, to complete the decision **before** proceeding to the actual preparation step.

Reply:
I got it!